app/models/validations/sales/setup_validations.rb

```diff
@@ -1,11 +1,43 @@
 module Validations::Sales::SetupValidations
   include Validations::SharedValidations
+  include CollectionTimeHelper
 
   def validate_saledate(record)
     return unless record.saledate && date_valid?("saledate", record)
 
-    unless record.saledate.between?(Time.zone.local(2022, 4, 1), Time.zone.local(2023, 3, 31)) || !FeatureToggle.saledate_collection_window_validation_enabled?
-      record.errors.add :saledate, I18n.t("validations.setup.saledate.financial_year")
+    unless record.saledate.between?(current_collection_start_date, active_collection_end_date)
+      record.errors.add :saledate, validation_error_message
+    end
+  end
+
+private
+
+  def active_collection_end_date
+    if FormHandler.instance.sales_in_crossover_period?
+      next_collection_end_date
+    else
+      current_collection_end_date
+    end
+  end
```

Comment:
If it's in crossover period, shouldn't we be returning previous form? and current if not in crossover?